network/src/lib.rs

```diff
@@ -10,22 +10,18 @@ pub use interface::NetworkProvider;
 
 pub mod connectivity_manager;
 pub mod constants;
+pub mod counters;
 pub mod error;
 pub mod interface;
 pub mod logging;
+pub mod noise;
```

Comment:
is this a pattern we want to avoid? By having it `pub` we signal that this is something that other crates here can use, whereas this is not true (besides fuzzing, but this is part of testing)

Reply:
hmm, i don't know what the right tradeoff is here, but I feel the conditional compilation is maybe getting a bit out of hand imo

Reply:
yep, pretty ugly :/ it'd be nice to have a macro for writing stuff like

```
#[pub(not(any(feature = "testing", feature = "fuzzing")))]
mod noise;
```